presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -537,6 +538,14 @@ private JoinEnumerationResult setJoinNodeProperties(JoinNode joinNode)
             if (isAtMostScalar(joinNode.getLeft(), lookup)) {
                 return createJoinEnumerationResult(joinNode.flipChildren().withDistributionType(REPLICATED));
             }
+
+            if (isBelowMaxBroadcastSize(joinNode, context) && isBelowMaxBroadcastSize(joinNode.flipChildren(), context)) {
+                JoinNode result = confidenceBasedBroadcast(joinNode, context);
```

Comment:
If this function is referenced in multiple files, consider find a util file for this function.